micrometer-core/src/main/java/io/micronaut/configuration/metrics/binder/web/ServerRequestMeterRegistryFilter.java

```diff
@@ -56,8 +59,15 @@ public ServerRequestMeterRegistryFilter(MeterRegistry meterRegistry) {
     }
 
     private String resolvePath(HttpRequest<?> request) {
-        Optional<String> route = request.getAttribute(URI_TEMPLATE, String.class);
-        return route.orElseGet(request::getPath);
+        Optional<Object> routeInfo = request.getAttribute(HttpAttributes.ROUTE_INFO)
```

Comment:
change this to `request.getAttribute(HttpAttributes.ROUTE_INFO, UriRouteMatch.class)` and you can avoid the filter and the cast and change the return type to `Optional<String>`